toolkit/tools/pkg/imagecustomizerlib/customizeutils_test.go

```diff
@@ -290,6 +291,198 @@ func TestAddCustomizerRelease(t *testing.T) {
 	assert.Equal(t, expectedDate, config["BUILD_DATE"])
 }
 
+func TestCustomizeImageSELinux(t *testing.T) {
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi)
+
+	testTmpDir := filepath.Join(tmpDir, "TestCustomizeImageSELinux")
+	buildDir := filepath.Join(testTmpDir, "build")
+	outImageFilePath := filepath.Join(testTmpDir, "image.raw")
+
+	// Customize image: SELinux enforcing.
+	// This tests enabling SELinux on a non-SELinux image.
+	configFile := filepath.Join(testDir, "selinux-force-enforcing.yaml")
+	err := CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "raw", "",
+		true /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Connect to customized image.
+	imageConnection, err := connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Verify bootloader config.
+	grubCfgFilePath := filepath.Join(imageConnection.Chroot().RootDir(), "/boot/grub2/grub.cfg")
+	grubCfgContents, err := file.Read(grubCfgFilePath)
+	assert.NoError(t, err, "read grub.cfg file")
+	assert.Regexp(t, "linux.* security=selinux ", grubCfgContents)
+	assert.Regexp(t, "linux.* selinux=1 ", grubCfgContents)
+	assert.Regexp(t, "linux.* enforcing=1 ", grubCfgContents)
+
+	// Verify SELinux config.
+	selinuxConfigPath := filepath.Join(imageConnection.Chroot().RootDir(), "/etc/selinux/config")
+	selinuxConfigContents, err := file.Read(selinuxConfigPath)
+	assert.NoError(t, err, "read SELinux config file")
+	assert.Regexp(t, "(?m)^SELINUX=enforcing$", selinuxConfigContents)
+
+	// Verify packages are installed.
+	ensureFilesExist(t, imageConnection, "/etc/selinux/targeted", "/var/lib/selinux/targeted/active/modules",
+		"/usr/bin/seinfo", "/usr/sbin/semanage")
+
+	err = imageConnection.CleanClose()
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Customize image: SELinux disabled.
+	// This tests disabling (but not removing) SELinux on an SELinux enabled image.
+	configFile = filepath.Join(testDir, "selinux-disabled.yaml")
+	err = CustomizeImageWithConfigFile(buildDir, configFile, outImageFilePath, nil, outImageFilePath, "raw", "",
+		true /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Connect to customized image.
+	imageConnection, err = connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Verify bootloader config.
+	grubCfgFilePath = filepath.Join(imageConnection.Chroot().RootDir(), "/boot/grub2/grub.cfg")
```

Comment:
nit: could this block of grub.cfg be shared

Reply:
Good call. Thanks. Fixed.